ophyd/tests/test_bluesky_protocols.py

```diff
@@ -1,93 +1,157 @@
+    else:
+        raise Exception(f"Unsupported type checking tool: '{tool_name}'")
+
+
+@pytest.mark.timeout(60)
+def test_configurable(run):
+    cmd = imports + "from bluesky.protocols import Configurable;"
+    run(cmd + "foo: Configurable = hw.motor1")
+    run(cmd + "foo: Configurable = ophyd.Device(name='test')")
+    run(cmd + "foo: Configurable = hw.signal")
+
+
+def test_triggerable(run):
+    cmd = imports + "from bluesky.protocols import Triggerable;"
+    run(cmd + "foo: Triggerable = hw.det")
+
+
+def test_checkable(run):
     cmd = imports + "from bluesky.protocols import Checkable;"
     run(cmd + "foo: Checkable = hw.motor1")
     run(cmd + "foo: Checkable = ophyd.Device(name='test')")
 
 
-def test_flyable():
+def test_hashints(run):
+    cmd = imports + "from bluesky.protocols import HasHints;"
+    run(cmd + "foo: HasHints = ophyd.Signal(name='test')")
+
+
+def test_flyable(run):
     cmd = imports + "from bluesky.protocols import Flyable;"
     run(cmd + "foo: Flyable = hw.flyer1")
     run(cmd + "foo: Flyable = sim.TrivialFlyer()")
 
 
-def test_hinted():
-    cmd = imports + "from bluesky.protocols import Hinted;"
-    run(cmd + "foo: Hinted = ophyd.Signal(name='test')")
-
-
-def test_movable():
+@pytest.mark.timeout(60)
+def test_movable(run):
     cmd = imports + "from bluesky.protocols import Movable;"
     run(cmd + "foo: Movable = hw.motor1")
-    run(cmd + "foo: Movable = ophyd.Device(name='test')")
     run(cmd + "foo: Movable = hw.flyer1")
+    run(cmd + "foo: Movable = ophyd.Component(ophyd.Signal, 'prefix')")
+    run(cmd + "foo: Movable = ophyd.Device(name='test')")
+    run(cmd + "foo: Movable = ophyd.Component(ophyd.Signal, 'prefix')")
 
 
-def test_pausable():
+def test_pausable(run):
     cmd = imports + "from bluesky.protocols import Pausable;"
     run(cmd + "foo: Pausable = hw.motor1")
     run(cmd + "foo: Pausable = ophyd.Device(name='test')")
 
 
-def test_readable():
+def test_readable(run):
     cmd = imports + "from bluesky.protocols import Readable;"
     run(cmd + "foo: Readable = hw.motor1")
     run(cmd + "foo: Readable = ophyd.Device(name='test')")
-    run(cmd + "foo: Readable = hw.signal1")
+    run(cmd + "foo: Readable = hw.signal")
 
 
-def test_stageable():
+def test_stageable(run):
     cmd = imports + "from bluesky.protocols import Stageable;"
     run(cmd + "foo: Stageable = hw.motor1")
     run(cmd + "foo: Stageable = ophyd.Device(name='test')")
 
 
-def test_status():
+def test_status(run):
     cmd = imports + "from bluesky.protocols import Status;"
     run(cmd + "foo: Status = ophyd.status.Status()")
     run(cmd + "foo: Status = ophyd.status.StatusBase()")
 
 
-def test_stoppable():
+def test_stoppable(run):
     cmd = imports + "from bluesky.protocols import Stoppable;"
     run(cmd + "foo: Stoppable = hw.motor1")
     run(cmd + "foo: Stoppable = hw.flyer1")
 
 
-def test_subscribable():
+# TODO: Ophyd signature is incompatible with bluesky protocol
```

Comment:
I have a feeling that Ophyd devices shouldn't conform to the `Subscribable` protocol as the concept of subscription is different between the two libraries.

The ophyd signature looks like this:
`def subscribe(self, callback, event_type=None, run=True) -> int`
and the Bluesky protocol looks like this:
`def subscribe(self, function: Callback[T]) -> None`

They seem to work differently in that Ophyd returns a `cid` when you subscribe which you can use as a token to unsubscribe. Bluesky doesn't give you a `cid` and unsubscribing works by passing the callback.

Reply:
Hmmm I don't know if the cid is useful here?

It's not like it would be accessible from the private [`_temp_callback_ids`](https://github.com/bluesky/bluesky/blob/5f5d2c1db8ee5c52fbc9a5f9e4fb3513b2d8625e/src/bluesky/run_engine.py#L753).

I'm not really sure what we need `event_type` for here either:

https://github.com/bluesky/bluesky/blob/5f5d2c1db8ee5c52fbc9a5f9e4fb3513b2d8625e/src/bluesky/suspenders.py#L67-L68

@tacaswell

Reply:
Now that I think about it, are the concepts even the same between Bluesky and Ophyd? In Ophyd a subscription is for watching for changes in device attributes, and in Bluesky a subscription is for handling events (e.g. with BestEffortCallback)

FWIW I know in our code base we have instances where we use `event_type` when subscribing to an Ophyd device to filter what events we want to receive and we store the CID to unsubscribe later.